src/mscorlib/src/System/Utf8String.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    // This is an experimental type and not referenced from CoreFx but needs to exists and be public so we can prototype in CoreFxLab.
+    public sealed class Utf8String
+    {
+        // Do not reorder these fields. Must match layout of Utf8StringObject in object.h.
+        private int _length;
+        [CLSCompliant(false)]
+        public byte _firstByte; // TODO: Is public for experimentation in CoreFxLab. Will be private in its ultimate form.
```

Comment:
Do you really need this to be public? It would be better to have `readonly byte GetPinnableReference()` method - that is the proper public method we will want to have eventually.

Reply:
It allows the code in CoreFxLab to be written as instance members of Utf8String itself would be. 

Though if `GetPinnableReference()` adds no overhead, I'm fine with it. This PR is unblock @GrabYourPitchforks so I'll let him make the call here.

Reply:
Sure, use `GetPinnableReference` instead if you wish. The only overhead I can think of is that an implicit null check may be emitted, but the JIT is generally good about eliding these where possible. If we find stray null checks we can open JIT bugs.